plugin/keyring_vault/vault_credentials_parser.h

```diff
@@ -23,21 +23,20 @@ namespace keyring
       optional_value.insert("vault_ca");
     }
 
-    my_bool parse(std::string *file_url, Vault_credentials *vault_credentials);
+    bool parse(const std::string &file_url, Vault_credentials *vault_credentials);
```

Comment:
Why do we need this class at all if we can put all its members inside `parse()` method as temporary local variables?
As far as I understood you never reuse the instance of this class and call 'parse()' several times.
I would leave just one static method / global function:
```c++
struct Vault_credentials_parse
{
   static bool parse(ILogger *logger, const std::string &file_url, Vault_credentials *vault_credentials);
}
```
or just
```c++
bool parse_vault_credentials(ILogger *logger, const std::string &file_url, Vault_credentials *vault_credentials);
```

Other helper functions can be hidden inside `.cpp` file